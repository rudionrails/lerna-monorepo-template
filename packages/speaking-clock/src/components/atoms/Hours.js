import React from "react";
import PropTypes from "prop-types";

const Hours = ({ hours, minutes }) => (
  <div className="font-weight-bold">
    {minutes === 0 ? hours : `${hours} hours`}
  </div>
);

Hours.propTypes = {
  hours: PropTypes.number.isRequired,
  minutes: PropTypes.number.isRequired,
};

export default Hours;
