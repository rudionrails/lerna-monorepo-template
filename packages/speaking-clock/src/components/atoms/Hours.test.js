import React from "react";

// module under test
import Hours from "./Hours";

const props = {
  hours: 13,
  minutes: 13,
};

test("to render hours", () => {
  const subject = shallow(<Hours {...props} />);
  expect(subject.html()).toContain("13 hours");
});

test("to render hours when minutes are zero", () => {
  const subject = shallow(<Hours {...props} minutes={0} />);
  expect(subject.html()).toContain("13");
});

test("to match snapshot", () => {
  const subject = shallow(<Hours {...props} />);
  expect(subject).toMatchSnapshot();
});
