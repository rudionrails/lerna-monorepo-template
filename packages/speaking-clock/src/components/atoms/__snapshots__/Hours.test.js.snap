// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snapshot 1`] = `
<div
  className="font-weight-bold"
>
  13 hours
</div>
`;
